// google-feed/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports.find = find;
exports.load = load;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _request = require('request');

var _request2 = _interopRequireDefault(_request);

/**
 * @param {String} query
 * @param {Object} options
 * @return {Promise}
 */

function find(query) {
  'use strict';

  var options = arguments[1] === undefined ? {} : arguments[1];
  return method('find', query, options);
}

/**
 * @param {String} query
 * @param {Object} options
 * @return {Promise}
 */

function load(query) {
  'use strict';

  var options = arguments[1] === undefined ? {} : arguments[1];
  return method('load', query, options);
}

/**
 * @param {String} method
 * @param {String} query
 * @param {Object} options
 * @return {Promise}
 */
function method(method, query, options) {
  'use strict';

  var defaultOptions = {
    q: query,
    output: 'json',
    v: '1.0'
  };

  return new Promise(function (resolve, reject) {
    (0, _request2['default'])({
      url: 'https://ajax.googleapis.com/ajax/services/feed/' + method,
      qs: Object.assign(options, defaultOptions),
      json: true
    }, function (error, response, body) {
      if (error) {
        reject(error);
      } else {
        resolve(body.responseData);
      }
    });
  });
}
